tests/infrastructure/prometheus.go

```diff
@@ -655,3 +601,37 @@ var _ = DescribeInfra("[rfe_id:3187][crit:medium][vendor:cnv-qe@redhat.com][leve
 		Entry("[test_id:6245] by IPv6", k8sv1.IPv6Protocol),
 	)
 })
+
+func countReadyAndLeaderPods(pod *k8sv1.Pod, component string) (foundMetrics map[string]int, err error) {
+	virtClient := kubevirt.Client()
+	target := fmt.Sprintf("virt-%s", component)
+	endpoint, err := virtClient.CoreV1().Endpoints(flags.KubeVirtInstallNamespace).Get(context.Background(), "kubevirt-prometheus-metrics", metav1.GetOptions{})
+	foundMetrics = map[string]int{
+		"ready":   0,
+		"leading": 0,
+	}
```

Comment:
nit: You could replace this now with two simple integers and return both + the error.

Not really a must.